irr/include/coreutil.h

```diff
@@ -75,9 +76,9 @@ inline io::path &deletePathFromFilename(io::path &filename)
 
 	if (p != s) {
 		++p;
-		filename = p;
+		res = p;
 	}
-	return filename;
+	return res;
```

Comment:
this is incorrect if `p == s`
but you can simplify it anyway:
```cpp
if (p != s)
    ++p;
return p;
```
and remove `res` above